src/App.js

```diff
@@ -2,41 +2,35 @@ import React from 'react';
 import './style.css';
 import Buttonpanel from './Components/buttonpanel';
 import Display from './Components/display';
-import calculate from './Logic/calculate';
-import calculator from './Logic/calculate';
+import Calculator from './Logic/calculate';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { result: '' };
+    this.state = { result: '', calculator: new Calculator() };
+    this.handleClick = this.handleClick.bind(this);
   }
-  handleClick = (name) => {
-    debugger;
+
+  handleClick(name) {
     const { calculator } = this.state;
-    if (!calculator) {
-      calculator.tap(name);
-      this.setState({ result: calculator.getnumber(), calculator: new calculate() })
-    } else {
-      calculator.tap(name);
-      this.setState({ result: calculator.getnumber(), calculator: calculator })
-    }
-  };
+    calculator.tap(name);
+    this.setState({ result: calculator.getnumber(), calculator });
+  }
```

Comment:
my above comment will affect this function

Reply:
🚀